java/mykola-melnyk/src/com/zoolatech/lecture2/tasks/_2/SimpleRobot.java

```diff
@@ -0,0 +1,226 @@
+package com.zoolatech.lecture2.tasks._2;
+
+/**
+ * Define a class that represents a robot, which moves in a room of a width W
+ * and a height of H. The class should provide methods to move a robot forward
+ * by either 1 or N tiles and turn it either left or right by 90 degrees from
+ * the current direction. The class should also provide methods to get the robot's
+ * current location and direction.
+ * The north-west corner of a room is represented by coordinates (0,0);
+ * north-east - (W-1, 0); south-east - (W-1, H-1); south-west - (0, H-1).
+ * By default, a robot should be placed in the north-west corner facing south direction.
+ * However, it should be also possible to give any coordinates and direction as the
+ * initial robot location. Robot must not be able to get outside a room - in
+ * such case the robot should move as much as possible to the room border, print
+ * a warning message and stop. All next moves in the same direction should be prevented.
+ */
+
+import java.util.Scanner;
+
+enum Direction {   //set direction variants
+    WEST,
+    EAST,
+    NORTH,
+    SOUTH
+}
+
+public class SimpleRobot {
+    private int x = 0; // this will be a position on x-axis (along width)
+    private int y = 0;  // this will be a position along y-axis (along height)
+    private int W;  // this will be a width value
+    private int H;  // this will be a height value
+    private Direction dir;  // we set a direction
+    private String compass; //to convert enum value to human-readable output
+
+
+    { //Initialization block
```

Comment:
The initialization should be done in the main method: all input data from a user should be passed to the object constructor.

Reply:
Resolved: rewrote the program code completely.